docs-v2/customize/guides/advanced/validate-input-and-output.mdx

```diff
@@ -0,0 +1,157 @@
+
+- **CLI**: When running a dry run using the Nango CLI, all your input and output will be automatically checked and will stop execution
+- **Production**: This file is stored along your scripts and execute the same check at execution time. Validation errors are only logged and do not halt execution.
+
+<Tip>
+The behavior in production ensures that validation issues are tracked without disrupting the application's operation.
+We are planning to enforce validation errors in the coming months with a way to configure the failure mode; a communication will be broadcasted before this goes into effects.
+</Tip>
+
+## Using `schema.json` in your codebase
+
+JSON Schema is supported in most of the main software languages, here is a non-exhaustive list of how you can directly use this file to validate the records you receive from Nango.
+
+<Tabs>
+
+<Tab title="Typescript">
+
+```ts
+import { Ajv } from 'ajv';
+import addFormats from 'ajv-formats';
+import jsonSchema from '.nango/schema.json';
+
+// Initiate AJV
+const ajv = new Ajv({ allErrors: true, discriminator: true });
+addFormats(ajv);
+
+const modelToValidate = 'MyModelName';
+const myData = {"id": "hello-word"};
+
+// Compile the JSON schema
+const validate = ajv.compile({
+  ...jsonSchema,
+  ...jsonSchema['definitions'][modelToValidate]
+});
+
+// Validate your data
+validate(mydata);
+```
+</Tab>
+
+
+<Tab title="Golang">
+```go
+package main
+
+import (
+	"fmt"
+	"log"
+	"github.com/santhosh-tekuri/jsonschema/v5"
+)
+
+func main() {
+  sch, err := jsonschema.Compile(".nango/schema.json")
+  if err != nil {
+    log.Fatalf("%#v", err)
+  }
+
+  myData := map[string]interface{}{
+		"id": "hello-word",
+	}
+  if err = sch.Validate(v); err != nil {
+    log.Fatalf("%#v", err)
+  }
+}
+```
+</Tab>
+
+<Tab title="Rust">
+```rust
+use jsonschema::{Draft, JSONSchema};
+use serde_json::json;
+
+let data = fs::read_to_string(".nango/schema.json").expect("Unable to read file");
+let schema: serde_json::Value = serde_json::from_str(&data).expect("Unable to parse");
+
+let instance = json!("{'id': 'hello-word'}");
+let compiled = JSONSchema::compile(&schema).expect("A valid schema");
+let result = compiled.validate(&instance);
+```
+</Tab>
+
+
+<Tab title="Python">
+```py
+import json
+from jsonschema import validate
+
+with open('.nango/schema.json') as file:
+    schema = json.load(file)
+    print(schema)
+
+    validate(instance={"id": "hello-word"}, schema=schema)
+```
+</Tab>
+</Tabs>
+
+
+
+
+## Custom Validation
+
+For more advanced use cases, you can generate your own validation schemas using the available files with the tool of your choice.
+
+
+
+<Tabs>
+
+<Tab title="Typescript `zod`">
+
+To create `zod` schemas you can use [`ts-to-zod` npm package](https://www.npmjs.com/package/ts-to-zod). This tool converts TypeScript definitions into Zod schemas, which can be used for runtime validation in your own Typescript codebase.
```

Comment:
I agree that it makes sense to explain that people can use this in their own code base. 

But I think we should add a section to explain that zod can be used directly in Nango scripts. I would introduce this sections before "Using `schema.json` in your codebase". Something like "Custom validation". This section should explain how to generate zod models, copy/paste them in another file so they can regenerate the original file, integrate this in their Nango scripts to perform more custom validation. 

This is an important use-case for Plain for example.